crates/core/app/tests/common/ibc_tests/mod.rs

```diff
@@ -0,0 +1,193 @@
+use {anyhow::Result, std::time::Duration};
+
+mod relayer;
+use anyhow::Context as _;
+use decaf377_rdsa::{SigningKey, SpendAuth, VerificationKey};
+use penumbra_app::{
+    app::{MAX_BLOCK_TXS_PAYLOAD_BYTES, MAX_EVIDENCE_SIZE_BYTES},
+    genesis,
+};
+use penumbra_keys::keys::{SpendKey, SpendKeyBytes};
+use penumbra_mock_consensus::TestNode;
+use penumbra_proto::core::component::stake::v1::Validator;
+use penumbra_shielded_pool::genesis::Allocation;
+use penumbra_stake::{DelegationToken, GovernanceKey, IdentityKey};
+#[allow(unused_imports)]
+pub use relayer::MockRelayer;
+
+mod node;
+pub use node::TestNodeWithIBC;
+use serde::Deserialize;
+use tendermint::{consensus::params::AbciParams, public_key::Algorithm, Genesis};
+
+/// Collection of all keypairs required for a Penumbra validator.
+/// Used to generate a stable identity for a [`NetworkValidator`].
+/// TODO: copied this from pd crate
```

Comment:
N.B. I've long wanted to pull out most of the network code from pd into a new crate `penumbra-network`, and in the process clean up the interfaces for wrangling the validator keyrings.